tempesta_fw/cache.c

```diff
@@ -805,17 +803,9 @@ tfw_cache_send_304(TfwHttpReq *req, TfwCacheEntry *ce)
 		return;
 	}
 
-	if (h_len > FRAME_MAX_LENGTH)
+	if (tfw_h2_make_frames(resp, stream_id, h_len, true, false))
```

Comment:
The `skip_body` argument for `tfw_h2_make_frames()` means that the function wouldn't frame body if it's present and caller function must somehow add frames by its own. On responses with empty body `skip_body` argument has no effect, since nothing to frame. I added a documentation to the function, but it seems insufficient, I will extend it.

Reply:
The "problem" is that `tfw_h2_make_frames()` executes some additional code under `!skip_body` - the code doesn't do much if there is no body at all, but there is still no sense to execute the code at all. Not big deal, but still worth fixing.

Reply:
The function was reworked.